subsys/net/lib/http/http_server_core.c

```diff
@@ -625,11 +632,35 @@ static int compare_strings(const char *s1, const char *s2)
 	return 1; /* Strings are not equal */
 }
 
+static bool skip_this(struct http_resource_desc *resource, bool is_websocket)
+{
+	struct http_resource_detail *detail;
+
+	detail = (struct http_resource_detail *)resource->detail;
+
+	if (is_websocket) {
+		if (detail->type != HTTP_RESOURCE_TYPE_WEBSOCKET) {
+			return true;
+		}
+	} else {
+		if (detail->type == HTTP_RESOURCE_TYPE_WEBSOCKET) {
+			return true;
+		}
+	}
+
+	return false;
```

Comment:
Maybe less readable, but can be a oneline.

```c
	return is_websocket != (detail->type == HTTP_RESOURCE_TYPE_WEBSOCKET);
```

Reply:
I would leave this as is, your suggestion is shorter but my brain just halts when trying to decipher the what is going on here.